lib/Doctrine/ORM/QueryBuilder.php

```diff
@@ -704,6 +704,10 @@ public function add($dqlPartName, $dqlPart, $append = false)
             }
 
             $dqlPart = $newDqlPart;
+
+            if (!$append) {
+                $isMultiple = false;
```

Comment:
Can you be a bit more specific about which part is confusing exactly?
Especially since i'm only using existing variables...